SU2_CFD/src/iteration_structure.cpp

```diff
@@ -664,13 +655,28 @@ void CFluidIteration::Solve(COutput *output,
 
     config[val_iZone]->SetInnerIter(Inner_Iter);
 
+    CSolverModule::for_each(*modules, CSolverModule::ActionPreIterationHook, output, config[val_iZone],
+                            geometry[val_iZone][val_iInst][MESH_0], solver[val_iZone][val_iInst][MESH_0]);
```

Comment:
Thats the call that iterates (at compile time) through all modules and calls their respective hooks.